src/NServiceBus.RabbitMQ/Connection/ChannelProvider.cs

```diff
@@ -8,14 +8,18 @@ class ChannelProvider : IChannelProvider, IDisposable
     {
         public ChannelProvider(ConnectionFactory connectionFactory, IRoutingTopology routingTopology, bool usePublisherConfirms)
         {
-            connection = new Lazy<IConnection>(() => connectionFactory.CreatePublishConnection());
-
+            this.connectionFactory = connectionFactory;
             this.routingTopology = routingTopology;
             this.usePublisherConfirms = usePublisherConfirms;
 
             channels = new ConcurrentQueue<ConfirmsAwareChannel>();
         }
 
+        public void CreateConnection()
```

Comment:
Something doesn't smell quite right about this, but I'm not sure what it is yet. However, I think it's good enough for the patch. I may get some crazy refactoring idea for `develop` later. 😉

Reply:
I'll be interested to see what you might come up with. We need some way to start the connection at endpoint start, but not sooner. The other places that get a `ConnectionFactory` all get called at endpoint start by core, so that's when those connections are opened.